delivery_carrier_typology/models/product.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2023 Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    prohibited_shipping_means_ids = fields.Many2many(
+        "delivery.carrier.typology",
+        string="Prohibited shipping means",
+    )
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        templates = super().create(vals_list)
+        for template, vals in zip(templates, vals_list):
+            if "prohibited_shipping_means_ids" in vals:
+                template.product_variant_ids.write(
+                    {
+                        "prohibited_shipping_means_ids": vals[
+                            "prohibited_shipping_means_ids"
+                        ]
+                    }
+                )
+        return templates
+
+    def write(self, vals):
+        res = super().write(vals)
+        if "prohibited_shipping_means_ids" in vals:
+            self.product_variant_ids.write(
+                {"prohibited_shipping_means_ids": vals["prohibited_shipping_means_ids"]}
+            )
+        return res
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    prohibited_shipping_means_ids = fields.Many2many(
+        "delivery.carrier.typology",
+        string="Prohibited shipping means",
+    )
```

Comment:
If we create new variants after adding the prohibited shipping mean on the product template, the news variants will not have the prohibited shipping mean.

Reply:
@Kev-Roche , it's fixed.